packages/plugin-ext/src/main/node/resolvers/plugin-local-file-resolver.ts

```diff
@@ -0,0 +1,56 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { PluginDeployerResolver, PluginDeployerResolverContext } from '../../../common/plugin-protocol';
+import { injectable } from 'inversify';
+import * as fs from 'fs';
+import * as path from 'path';
+import { FileUri } from '@theia/core/lib/node';
+import URI from '@theia/core/lib/common/uri';
+
+@injectable()
+export class LocalFilePluginDeployerResolver implements PluginDeployerResolver {
+
+    static LOCAL_FILE = 'local-file';
```

Comment:
should be only `file` no ?

file:///path-to-file

Reply:
That was also my first thought, but I decided to mirror `local-dir:`, which expects a filesystem path after the prefix. Using `file://` would for imply an URI for me, which this is currently not.

Reply:
In addition, if we support the generic `file://` scheme I would expect that this works for files as well as directories

Reply:
You're giving
`local-file:${URI.revive(vsixUriOrExtensionId).fsPath}`

so it looks like a path no ?

And you can say that `file://` only supports file, not directories

Reply:
Yes, it's a path, but a path is not an URI. They have different rules for encoding stuff and mixing them up usually leads to weird issues.

So I would rather prefer to stay with `local-file` because it makes it pretty clear that it has the same format as `local-dir`, just referring to files instead of directories. 

If necessary, we could still add support for the `file` scheme in a separate MR but this should then be taking real URIs and be able to handle both files and directories.

Would that be OK for you?

Reply:
yes I approved PR yesterday